lint/settings.py

```diff
@@ -0,0 +1,304 @@
+        prefs.add_on_change('sublimelinter-pref-settings', observer or self.on_prefs_update)
+
+    def observe(self, observer=None):
+        """Observer changes to the plugin settings."""
+        self.plugin_settings = sublime.load_settings(SETTINGS_FILE)
+        self.plugin_settings.clear_on_change('sublimelinter-persist-settings')
+        self.plugin_settings.add_on_change('sublimelinter-persist-settings',
+                                           observer or self.on_update)
+
+    def get_merged_settings(self):
+        """Returns dict of default and user settings merged."""
+        res = sublime.find_resources(SETTINGS_FILE)
+        merged_dict = {}
+        for r in res:
+            s = sublime.load_resource(r)
+            d = sublime.decode_value(s)
+            merged_dict.update(d)
+        return merged_dict
+
+    def on_update_call(self, callback):
+        """Set a callback to call when user settings are updated."""
+        self.on_update_callback = callback
+
+    def on_update(self):
+        """
+        Update state when the user settings change.
+
+        The settings before the change are compared with the new settings.
+        Depending on what changes, views will either be redrawn or relinted.
+
+        """
+
+        self.settings = self.get_merged_settings()
+
+        self.changeset.extend(self.dict_comparer(self.settings))
+
+        print("self.changeset: ", self.changeset)
+
+        from . import persist
+
+        # TODO: or force_xml_scheme
+        if not self.changeset:
+            return
+
+        if "force_xml_scheme" in self.changeset:
+            persist.printf("Scheme mode changed regenerating style definitions.")
+
+            from . import persist
+            persist.scheme.generate()
+
+        if "styles" in self.changeset:
+            persist.printf("Style definitions changed. Regenerating.")
+            persist.scheme.generate()
+
+        # Clear the path-related caches if the paths list has changed
+        if "paths" in self.changeset:
+            util.clear_path_caches()
+
+        # Add python paths if they changed
+        if "python_paths" in self.changeset:
+            python_paths = self.settings.get('python_paths', {}).get(
+                sublime.platform(), [])
+
+            for path in python_paths:
+                if path not in sys.path:
+                    sys.path.append(path)
+
+        # If the syntax map changed, reassign linters to all views
+        from .linter import Linter
+
+        if "syntax_map" in self.changeset:
+            Linter.clear_all()
+            util.apply_to_all_views(lambda view: Linter.assign(view, reset=True))
+
+        if "gutter_theme" in self.changeset:
+            self.update_gutter_marks()
+
+        # TODO: change whether this optimal:
+        Linter.reload()  # always reload
+        from ..sublime_linter import SublimeLinter
+        SublimeLinter.lint_all_views()
+
+        # TODO: what does this do?
```

Comment:
I dunno man. Just remove it and we'll dig it out of the git log or whatever if something breaks.